skrub/_fixes.py

```diff
@@ -0,0 +1,19 @@
+import sklearn
+from sklearn.utils.fixes import parse_version
+
+sklearn_version = parse_version(parse_version(sklearn.__version__).base_version)
+
+
+if sklearn_version < parse_version("1.6"):
+    from sklearn.utils._tags import _safe_tags as get_tags  # noqa
+else:
+    from sklearn.utils import get_tags  # noqa
+
+
+def _check_n_features(estimator, X, *, reset):
```

Comment:
Should we monkey patch sklearn's `BaseEstimator` with this function in skrub instead of calling it explicitly from `skrub._fixes`?

Reply:
Doing the monkey patch is undoing the change done in scikit-learn. So it means that we are going to have our own version of `BaseEstimator` in skrub. I think this is safer to implement our own `_check_n_features` and drop it at the time that the minimum supported version of scikit-learn is 1.6, we switch back to the scikit-learn version. It looks less invasive.

This is already a kind of pattern that we have in `scikit-learn` when it comes to numpy or scipy (cf. `sklearn/utils/fixes.py`).